includes/ProductSync/ProductValidator.php

```diff
@@ -107,6 +107,24 @@ public function __construct( WC_Facebookcommerce_Integration $integration, WC_Pr
 		$this->integration      = $integration;
 	}
 
+	/**
+	 * __get method for backward compatibility.
+	 *
+	 * @param string $key property name
+	 * @return mixed
+	 * @since x.x.x
+	 */
+	public function __get( $key ) {
+		// Add warning for private properties.
+		if ( 'facebook_product' === $key ) {
+			/* translators: %s property name. */
+			_doing_it_wrong( __FUNCTION__, sprintf( esc_html__( 'The %s property is private and should not be accessed outside its class.', 'facebook-for-woocommerce' ), esc_html( $key ) ), 'x.x.x' );
```

Comment:
Again `facebook_product` is `protected`.

Reply:
Thank you for bringing this to my notice. I have fixed it in the [recent commit](https://github.com/woocommerce/facebook-for-woocommerce/pull/2619/commits/cd911a8ac92c716106bd5886a92c550caa797f92).